plugins/gitops-profiles/src/components/ClusterPage/ClusterPage.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import {
+  Content,
+  ContentHeader,
+  Header,
+  SupportButton,
+  Page,
+  pageTheme,
+  Table,
+  Progress,
+  HeaderLabel,
+  useApi,
+} from '@backstage/core';
+import ClusterTable from '../ClusterTable/ClusterTable';
+import { Button, Link, Typography } from '@material-ui/core';
+import { useParams } from 'react-router-dom';
+import { useAsync, useLocalStorage } from 'react-use';
+import { gitOpsApiRef, ListClusterStatusesResponse, Status } from '../../api';
+import { transformRunStatus } from '../ProfileCatalog';
+
+const ClusterPage: FC<{}> = () => {
+  const params = useParams<{ owner: string; repo: string }>();
+
+  const [loginInfo] = useLocalStorage<{
+    token: string;
+    username: string;
+    name: string;
+  }>('githubLoginDetails');
+  const [pollingLog, setPollingLog] = useState(true);
+  const [runStatus, setRunStatus] = useState<Status[]>([]);
+  const [runLink, setRunLink] = useState<string>('');
+  const [showProgress, setShowProgress] = useState(true);
+
+  const api = useApi(gitOpsApiRef);
+
+  const columns = [
+    { field: 'status', title: 'Status' },
+    { field: 'message', title: 'Message' },
+  ];
+
+  useEffect(() => {
+    if (pollingLog) {
+      const interval = setInterval(async () => {
+        const resp = await api.fetchLog({
+          gitHubToken: loginInfo.token,
+          gitHubUser: loginInfo.username,
+          targetOrg: params.owner,
+          targetRepo: params.repo,
+        });
+
+        setRunStatus(resp.result);
+        setRunLink(resp.link);
+        if (resp.status === 'completed') {
+          setPollingLog(false);
+          setShowProgress(false);
+        }
+      }, 10000);
+      return () => clearInterval(interval);
+    }
+    return () => {};
+  }, [pollingLog]);
+
+  if (params.owner === undefined || params.repo === undefined) {
+    const { loading, error, value } = useAsync<ListClusterStatusesResponse>(
```

Comment:
Can't call this hook conditionally

Reply:
could you collaborate this?

Reply:
You shouldn't call hooks in conditional code, see plugins/auth-backend/src/providers/OAuthProvider.test.ts

Reply:
acked

Reply:
fixed and I had to refactor ClusterPage component into two to avoid the if.

Reply:
thank you for the comment, @Rugvip!